internal/events/event_consumer_test.go

```diff
@@ -78,40 +69,148 @@ func Test_EventConsumer_Consume_SendDLQ(t *testing.T) {
+			if len(m) != 1 {
+				return false
+			}
+			message := m[0]
+			// Verify that there is only 1 successful message
+			if len(message.SuccessfulExecutions) != 1 {
+				return false
+			}
+
+			// Verify that one error is recorded correctly. in 3 seconds there should be one backoff.
+			if len(message.Errors) != 1 && message.Errors[0].ErrorMessage != handlingErr.Error() {
+				return false
+			}
+
+			// Verify that re-broadcasted message is the same as the one that failed
+			return message.Key == msg.Key && message.Topic == msg.Topic
+		})).
+		Return(nil)
+
+	getEntries := log.DefaultLogger.StartTest(log.InfoLevel)
+
+	ec.Consume(ctx)
+
+	entries := getEntries()
+	assert.Equal(t, "Starting consuming messages for topic test.test_topic...", entries[0].Message)
+	assert.Equal(t, log.InfoLevel, entries[0].Level)
+
+	assert.Equal(t, "Reading message from topic test.test_topic...", entries[1].Message)
+	assert.Equal(t, log.InfoLevel, entries[1].Level)
+
+	// logged by mock handler
+	assert.Equal(t, "Handling message with key key-1 by handler FailedEventHandler", entries[2].Message)
+	assert.Equal(t, log.InfoLevel, entries[2].Level)
+
+	// logged by mock handler
+	assert.Equal(t, "Handling message with key key-1 by handler SuccessfulEventHandler", entries[3].Message)
+	assert.Equal(t, log.InfoLevel, entries[3].Level)
+
+	assert.Equal(t, "Waiting 2s before retrying handling message with key key-1", entries[4].Message)
+	assert.Equal(t, log.WarnLevel, entries[4].Level)
+
+	assert.Equal(t, "Retrying handling message with key key-1", entries[5].Message)
+	assert.Equal(t, log.WarnLevel, entries[5].Level)
+	assert.Equal(t, entries[4].Time.Truncate(time.Second).Add(time.Second*2), entries[5].Time.Truncate(time.Second))
+
+	assert.Equal(t, "Handling message with key key-1 by handler FailedEventHandler", entries[6].Message)
+	assert.Equal(t, log.InfoLevel, entries[6].Level)
+
+	assert.Equal(t, "Handler SuccessfulEventHandler has already been executed for message with key key-1. Skipping...", entries[7].Message)
+	assert.Equal(t, log.InfoLevel, entries[7].Level)
+
+	assert.Equal(t, "Waiting 4s before retrying handling message with key key-1", entries[8].Message)
+	assert.Equal(t, log.WarnLevel, entries[8].Level)
+
+	assert.Equal(t, "Stopping consuming messages for topic test.test_topic due to context cancellation...", entries[9].Message)
+	assert.Equal(t, log.InfoLevel, entries[9].Level)
+
+	assert.Equal(t, "Replaying message with key key-1 to topic test.test_topic", entries[10].Message)
+	assert.Equal(t, log.WarnLevel, entries[10].Level)
+
+	consumerMock.AssertExpectations(t)
+	crashTrackerMock.AssertExpectations(t)
 }
 
 func Test_EventConsumer_Consume_FinalizeConsumer(t *testing.T) {
 	// setup mocks
 	consumerMock := &MockConsumer{}
 	crashTrackerMock := &crashtracker.MockCrashTrackerClient{}
-	producerMock := &MockProducer{}
+	producerMock := NewMockProducer(t)
+	failedEventHandlerMock := NewMockEventHandler(t)
```

Comment:
When you use `NewMockProducer` or `NewMockEventHandlers`, you don't need to assert expectations at the end, since the constructors handle the expectation assertions:

https://github.com/stellar/stellar-disbursement-platform-backend/blob/03fc782e1167f157dfa6059a0d8184c0c71ea711/internal/events/mocks.go#L69

Reply:
Nevermind, you've removed it already. Thanks!